stream_alert/alert_processor/outputs.py

```diff
@@ -0,0 +1,298 @@
+            headers [dict]: A dictionary containing header parameters
+
+        Returns:
+            [integer] ID of the Phantom container where the alerts will be sent
+        """
+        message = "StreamAlert Rule Triggered - {}".format(rule_name)
+        ph_container = {'name' : message,
+                        'description' : message}
+        container_string = json.dumps(ph_container)
+        resp = self._request_helper(container_url, container_string, headers, False)
+
+        if not self._check_http_response(resp):
+            self._log_status(False)
+            return False
+
+        try:
+            resp_dict = json.loads(resp.read())
+        except ValueError as err:
+            logging.error('An error occurred while decoding message to JSON: %s', err)
+            return False
+
+        return resp_dict and resp_dict['id']
+
+    def dispatch(self, descriptor, rule_name, alert):
+        """Send alert to Phantom
+
+        Args:
+            descriptor [string]: Service descriptor (ie: slack channel, pd integration)
+            rule_name [string]: The name of the triggered rule
+            alert [dict]: The alert relevant to the triggered rule
+        """
+
+        creds = self._load_creds(descriptor)
+        if not creds:
+            self._log_status(False)
+            return
+
+        headers = {"ph-auth-token": creds['ph_auth_token']}
+        container_url = os.path.join(creds['url'], 'rest/container/')
+        container_id = self._setup_container(rule_name, container_url, headers)
+
+        success = False
+        if container_id:
+            artifact = {"cef" : alert['record'],
+                        "container_id" : container_id,
+                        "data" : alert,
+                        "name" : "Phantom Artifact",
+                        "label" : "Alert"}
+            artifact_string = json.dumps(artifact)
+            artifact_url = os.path.join(creds['url'], 'rest/artifact/')
+            resp = self._request_helper(artifact_url, artifact_string, headers, False)
+
+            success = self._check_http_response(resp)
+
+        self._log_status(success)
+
+
+@streamoutput
+class SlackOutput(OutputBase):
+    """SlackOutput handles all alert dispatching for Slack"""
+    __service__ = 'slack'
+
+    @classmethod
+    def get_user_defined_properties(cls):
+        """Get properties that must be asssigned by the user for a Slack integration
+
+        Returns:
+            [OrderedDict] Contains various OutputProperty items
+        """
+        return OrderedDict([
+            ('descriptor',
+             OutputProperty(description='a short and unique descriptor for this Slack integration'
+                                        '(ie: channel, group, etc)')),
+            ('url',
+             OutputProperty(description='the full Slack webhook url, including the secret',
+                            cred_requirement=True))
+        ])
+
+    def dispatch(self, descriptor, _, alert):
```

Comment:
should we just use kwargs instead of negating certain positional arguments with `_`?